ripser/ripser.py

```diff
@@ -37,6 +37,82 @@
 from pyRipser import doRipsFiltrationDMSparse as DRFDMSparse
 
 
+def get_greedy_perm_pc(X, N = -1, metric="euclidean"):
+    """
+    Compute a furthest point sampling permutation of a set
+    of points
+    Parameters
+    ----------
+    X: ndarray(N, d)
+        Point cloud with N points in d dimensions
+    N: int
+        Number of points to take in the permutation
+    metric: string or callable
+        The metric to use when calculating distance between instances in a 
+        feature array. If metric is a string, it must be one of the options 
+        specified in PAIRED_DISTANCES, including "euclidean", "manhattan", 
+        or "cosine". Alternatively, if metric is a callable function, it is 
+        called on each pair of instances (rows) and the resulting value 
+        recorded. The callable should take two arrays from X as input and 
+        return a value indicating the distance between them.
+    Returns
+    -------
+    perm: ndarray(N)
+        Indices of points in the greedy permutation
+    lambdas: ndarray(N)
+        Covering radii at different points
+    """
+    if N == -1:
+        N = X.shape[0]
+    #By default, takes the first point in the list to be the
+    #first point in the permutation, but could be random
+    perm = np.zeros(N, dtype=np.int64)
+    lambdas = np.zeros(N)
+    ds = pairwise_distances(X, X[0, :][None, :], metric=metric)
+    for i in range(1, N):
+        idx = np.argmax(ds)
+        perm[i] = idx
+        lambdas[i-1] = ds[idx]
+        ds = np.minimum(ds, pairwise_distances(X, X[idx, :][None, :], metric=metric))
+    lambdas[-1] = np.max(ds)
+    return (perm, lambdas)
+
+
+
+def get_greedy_perm_dm(D, N = -1):
+    """
```

Comment:
This function to the previous one except 2 lines. Can't we combine them into one?